src/hooks/useStorage.js

```diff
@@ -1,9 +1,20 @@
 import { useEffect, useState } from "react";
 
+function objectOrString(possibleJson) {
+  if (!possibleJson) {
+    return possibleJson;
+  }
+  try {
+    return JSON.parse(possibleJson);
+  } catch (e) {
+      return possibleJson;
+  }
+}
+
 function useStorage(key, initialValue) {
   const [value, setValue] = useState(() => {
     const value = window.localStorage.getItem(key);
-    return value === null ? initialValue : value;
+    return value === null ? initialValue : objectOrString(value);
```

Comment:
This try-catch seems implicit and a source for bugs. I would prefer if you called `JSON.stringify()` and `JSON.parse()` directly, outside of this hook, if you need to serialize JSON.